### PR TITLE
Added On The Right Track Q2 2012

### PR DESCRIPTION
**Pull request by yud08**
plz rev

### CODE REVIEW

File: solutions/bio1/OnTheRightTrack/index.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { SCodeBlock, SSubtitle, SText, STitle, SCode, } from "components";
+
+export const OnTheRightTrack = () => {
+  return (
+    <>
+      <STitle>Idea</STitle>
+      <SText>
+        This is a quite straightforward implementation problem, and with the small-ish n $(n {'<'}= 10,000)$, a straightforward simulation is in order.
```
Comment (myst-6) [inline solutions/bio1/OnTheRightTrack/index.tsx:8]: do `n \le 10\,000` please

File: solutions/bio1/OnTheRightTrack/index.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      </SText>
+
+      <STitle>Solution</STitle>
+      <SText>
+        With larger problems (often Question $2$), it is useful to define a set of classes, and some methods embedded in them to help solve the problem.
+        In this problem, we can define a class, <SCode>Stop</SCode>, which will store the type of stop (lazy or flip flop) it is, which stops it is connected to, and which curved track the next train going through the straight one will exit through.
+        We will also update that specific curved track as each train passes through.
+        With this <SCode>Stop</SCode> class, tracking the train's movement is easy, and we can do this using a map which will store each stop indexed by it's character($A, B$, etc).
```
Comment (myst-6) [inline solutions/bio1/OnTheRightTrack/index.tsx:16]: "it's" should be "its"

also please add a space between "character" and the brackets

File: solutions/bio1/OnTheRightTrack/index.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        A helpful tip, when you use a map with a custom struct/class, be sure to implement a constructor which takes no parameters (which can be empty).
```
Comment (myst-6) [inline solutions/bio1/OnTheRightTrack/index.tsx:17]: The first comma should be a colon instead. Also, mention that this is a C++-specific tip.

File: solutions/bio1/OnTheRightTrack/index.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        This is because the map uses this to create a default value when no value is found for a key, which shouldn't be a problem in our implementation, but is needed for the map to function.
+        An unordered_map will require you to create an entire hash function for your class, which is considerably more tedious.
```
Comment (myst-6) [inline solutions/bio1/OnTheRightTrack/index.tsx:19]: unordered_map should be `<SCode>unordered_map</SCode>`

File: solutions/bio1/OnTheRightTrack/index.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        We will also update that specific curved track as each train passes through.
+        With this <SCode>Stop</SCode> class, tracking the train's movement is easy, and we can do this using a map which will store each stop indexed by it's character($A, B$, etc).
+        A helpful tip, when you use a map with a custom struct/class, be sure to implement a constructor which takes no parameters (which can be empty).
+        This is because the map uses this to create a default value when no value is found for a key, which shouldn't be a problem in our implementation, but is needed for the map to function.
+        An unordered_map will require you to create an entire hash function for your class, which is considerably more tedious. 
+      </SText>
+
+      <SSubtitle>Initialising the board</SSubtitle>
+      <SText>
+        The board is created quite strangely, but it only has $26$ stops, so making it is relatively easy, just be careful to link the stops correctly with the left and right stops.
```
Comment (myst-6) [inline solutions/bio1/OnTheRightTrack/index.tsx:24]: I think the last comma should be a semicolon.

**Review by myst-6 — CHANGES_REQUESTED** [review]
See the comments. Also, I think you need to add a Python translation for BIO1 questions (please could you confirm this @macaquedev ?)

File: solutions/bio1/OnTheRightTrack/index.tsx (hunk 4 above)
Comment (macaquedev) [reply solutions/bio1/OnTheRightTrack/index.tsx:19]: and again make sure it's clear that's C++ - specific

**Comment by macaquedev** [comment]
for BIO1 questions I think:

Q1/Q3 - Python and C++ MUST
Q2 - Python MUST C++ good to have but not necessary

just because solving those q2's is so so so much harder in C++.

Also, if someone has sols to those q2s which don't currently have c++ sols we can just add them later.
@myst-6 @yud08

**Comment by yud08** [comment]
ok impl

File: solutions/bio1/OnTheRightTrack/index.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        We will also update that specific curved track as each train passes through.
+        With this <SCode>Stop</SCode> class, tracking the train's movement is easy, and we can do this using a map which will store each stop indexed by its character ($A, B$, etc).
+        A helpful tip (specifically for C++): when you use a <SCode>map</SCode> with a custom struct/class, be sure to implement a constructor which takes no parameters (which can be empty).
+        This is because the map uses this to create a default value when no value is found for a key, which shouldn't be a problem in our implementation, but is needed for the map to function.
+        An <SCode>unordered_map</SCode> will require you to create an entire hash function for your class, which is considerably more tedious. 
+      </SText>
+
+      <SSubtitle>Initialising the board</SSubtitle>
+      <SText>
+        The board is created quite strangely, but it only has $26$ stops, so making it is relatively easy: just be careful to link the stops correctly with the left and right stops.
```
Comment (myst-6) [inline solutions/bio1/OnTheRightTrack/index.tsx:24]: The last comma should be a semicolon. Here's why:

The sentence contains two independent clauses, each of which could stand alone as a complete sentence.
A semicolon is used to link two related independent clauses without using a conjunction like "and" or "but."
So, the sentence should read:

"The board is created quite strangely, but it only has $26$ stops, so making it is relatively easy; just be careful to link the stops correctly with the left and right stops."

**Review by myst-6 — CHANGES_REQUESTED** [review]
e

File: solutions/bio1/OnTheRightTrack/index.tsx (hunk 6 above)
Comment (macaquedev) [reply solutions/bio1/OnTheRightTrack/index.tsx:24]: @myst-6 in his nerdy English teacher phase, and I'm here for it

**Comment by yud08** [comment]
merge 😭

**Comment by macaquedev** [comment]
@yud08 missing latex! 
image

**Review by macaquedev — CHANGES_REQUESTED** [review]
e

**Comment by yud08** [comment]
merge 🙏

**Review by myst-6 — DISMISSED** [review]
I think the meow is woof

**Comment by macaquedev** [comment]
Cool! @yud08 please fix merge conflicts and I'll accept!

**Review by macaquedev — APPROVED** [review]
good :)

**Pull request merged**